src/messages/MOSDOp.h

```diff
@@ -242,14 +236,14 @@ struct ceph_osd_request_head {
       ::encode_nohead(snaps, payload);
     } else {
       header.version = HEAD_VERSION;
-      ::encode(client_inc, payload);
+      ::encode(pgid, payload);
```

Comment:
Unfortunately we can't just reorder stuff like this. That's fine for testing changes but not for merging into upstream master. For merging you'll need to set a different version of the message and a feature bit so the different endpoints know which version to use. :/